### ISSUE TITLE
x/interchainstaking: Keeper.SetValidatorForZone should also check for tombstoned validators after retrieving the validator from store

### ISSUE THREAD

**Issue by odeke-em** [quicksilver-zone/quicksilver#608]
## Summary of Bug
If you read through the code for Keeper.SetValidatorForZone
https://github.com/quicksilver-zone/quicksilver/blob/c26b22e7cf2eebef24f70289f73552f54da807ea/x/interchainstaking/keeper/keeper.go#L291-L348

we only seem to be checking if a validator wasn't jailed, but I suggest that for found validators let's also explicitly check that they weren't tombstoned because even if it were jailed and could unjail itself, tombstoning is an infinite suspension time

## Suggestion
```go
} else {
   if val.Tombstoned {
       k.Logger(ctx).Error("Tombstoned validator found", "valoper", validator.OperatorAddress)
       return fmt.Errorf("%q on chainID: %q was found to already have been tombstoned", validator.OperatorAddress, zone.ChainID)
   }

   if !val.Jailed && validator.IsJailed() {
       ...
   }
   ...
}
```

/cc @elias-orijtech  

## Version
13bf586baffd07ca4a2a7f093212455fafb5df27

____

#### For Admin Use

- [x] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by odeke-em** [comment]
Also perhaps Keeper.SetValidatorsForZone should also immediately check for tombstoned validators to avoid setting validators for a zone that are invalid.

**Comment by ThanhNhann** [comment]
I think we just need to check `tombstoned` in `Keeper.SetValidatorsForZone` is enough because it will call `Keeper.SetValidatorForZone` after in callback

**Comment by joe-bowman** [comment]
We still want to have knowledge of the validator here (the entire valset is track, not just active validators), but we should adequately check and record the tombstone status too. @ThanhNhann is tracking this in #640

**Comment by ThanhNhann** [comment]
Solved with #642
